### PR TITLE
[winesteam] Add 64-bit prefix support

### PR DESCRIPTION
**Pull request by daniel-j**
With this patch, you can set the architecture in winesteam lutris installer scripts for games that require 64-bit.
```
game:
  appid: 12345
  arch: win64
```
It will create the win64 wineprefix `$HOME/.local/share/lutris/runners/winesteam/prefix64` and run steam from the 32-bit install, so you don't have to have two steam installations on disk.

This seem to require a system-install of Wine (64-bit obviously), because else Lutris tries to run it with it's default 32-bit wine, but that's related to the Wine runner.

Shoutout to **[DK] derklempner** who brought this to attention!

### CODE REVIEW

File: lutris/runners/winesteam.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def appid(self):
 
     @property
     def prefix_path(self):
-        _prefix = self.game_config.get('prefix') or self.get_default_prefix()
+        _prefix = self.game_config.get('prefix') or self.get_or_create_default_prefix(arch=self.game_config.get('arch'))
```
Comment (Freso) [inline lutris/runners/winesteam.py]: Does Lutris (aspire to) adhere to the PEP 8 line length limit? If so, this line needs to be split.

File: lutris/runners/winesteam.py [hunk 2]
```diff
@@ -156,7 +156,8 @@ def launch_args(self):
         args = [self.get_executable(), self.get_steam_path()]
 
         # Fix invisible text in Steam
-        args.append('-no-dwrite')
+        # Since Wine 1.7.50, DirectWrite has been fixed
+        # args.append('-no-dwrite')
```
Comment (Freso) [inline lutris/runners/winesteam.py]: If so, just remove the code entirely. (This should be separate commit though, so you can explain the reasoning properly.) The code will still be in the git repository, so there's no need to leave commented out stale code in the codebase.

File: lutris/runners/winesteam.py [hunk 3]
```diff
@@ -300,30 +301,34 @@ def get_default_steamapps_path(self):
...
         # Fix steam text display
+        # Since Wine 1.7.50, DirectWrite has been fixed.
         set_regedit("HKEY_CURRENT_USER\Software\Valve\Steam",
-                    'DWriteEnable', '0', 'REG_DWORD',
+                    'DWriteEnable', '1', 'REG_DWORD',
                     wine_path=self.get_executable(),
                     prefix=prefix_dir)
 
-    def get_default_prefix(self):
+    def get_default_prefix(self, arch=None):
```
Comment (Freso) [inline lutris/runners/winesteam.py]: Maybe explicitly set `'win32'` as the default (like for `get_or_create_default_prefix()`)? 64-bit is the default for more and more things, so explicitly declaring the current default here and handling both/all cases explicitly in the code would mean that whenever Lutris decides to switch to 64-bit Wine by default, it'll only have to update this line. (Though the argument is transferred over straight from `get_or_create_default_prefix()`, so this would also mean maintained the default both places…)

**Comment by daniel-j** [comment]
This fixes #430! Yay

**Comment by Freso** [comment]
I would [squash](https://duckduckgo.com/?q=git+squash) the two commits and add what you wrote in the description here to the new commit, as well as a line akin to "`Fixes https://github.com/lutris/lutris/issues/430`". This will assure that the arguments and reasoning for the changes survive the git repository, even if GitHub disappears. :) (The the "Fixes" line will make the issue auto-close when the PR gets merged into master.)

(See also https://chris.beams.io/posts/git-commit/ )

**Comment by strycore** [comment]
> This seem to require a system-install of Wine (64-bit obviously), because else Lutris tries to run it with it's default 32-bit wine, but that's related to the Wine runner.

We ship 64bit builds of every Wine version (including Staging), the Wine version can be specified in the install scripts in the runner section.
